AK no longer builds on OpenBSD

AK no longer builds as of 3bf77f01a7 by @ADKaster:

```
-- The C compiler identification is Clang 10.0.1
-- The CXX compiler identification is Clang 10.0.1
```

```
[65/2375] Building CXX object AK/Tests/CMakeFiles/TestIndexSequence.dir/TestIndexSequence.cpp.o
FAILED: AK/Tests/CMakeFiles/TestIndexSequence.dir/TestIndexSequence.cpp.o 
/usr/bin/c++   -I../Libraries -I../. -Wno-unknown-warning-option -Wall -Wextra -Werror -Wmissing-declarations -Wformat=2 -fdiagnostics-color=always -ftls-model=initial-exec -Wno-overloaded-virtual -std=c++2a -MD -MT AK/Tests/CMakeFiles/TestIndexSequence.dir/TestIndexSequence.cpp.o -MF AK/Tests/CMakeFiles/TestIndexSequence.dir/TestIndexSequence.cpp.o.d -o AK/Tests/CMakeFiles/TestIndexSequence.dir/TestIndexSequence.cpp.o -c ../AK/Tests/TestIndexSequence.cpp
In file included from ../AK/Tests/TestIndexSequence.cpp:29:
.././AK/TypeList.h:65:47: error: use 'template' keyword to treat 'Type' as a dependent template name
    (forward<F>(f)(TypeWrapper<typename List::Type<Indexes>> {}), ...);
                                              ^
                                              template 
.././AK/TypeList.h:77:48: error: use 'template' keyword to treat 'Type' as a dependent template name
    (forward<F>(f)(TypeWrapper<typename ListA::Type<Indexes>> {}, TypeWrapper<typename ListB::Type<Indexes>> {}), ...);
                                               ^
                                               template 
.././AK/TypeList.h:77:95: error: use 'template' keyword to treat 'Type' as a dependent template name
    (forward<F>(f)(TypeWrapper<typename ListA::Type<Indexes>> {}, TypeWrapper<typename ListB::Type<Indexes>> {}), ...);
                                                                                              ^
                                                                                              template 
3 errors generated.

```

--- a/AK/TypeList.h
+++ b/AK/TypeList.h
@@ -62,7 +62,7 @@ struct TypeWrapper {
 template<typename List, typename F, unsigned... Indexes>
 constexpr void for_each_type_impl(F&& f, IndexSequence<Indexes...>)
 {
-    (forward<F>(f)(TypeWrapper<typename List::Type<Indexes>> {}), ...);
+    (forward<F>(f)(TypeWrapper<typename List::template Type<Indexes>> {}), ...);
 }
 
 template<typename List, typename F>
@@ -74,7 +74,7 @@ constexpr void for_each_type(F&& f)
 template<typename ListA, typename ListB, typename F, unsigned... Indexes>
 constexpr void for_each_type_zipped_impl(F&& f, IndexSequence<Indexes...>)
 {
-    (forward<F>(f)(TypeWrapper<typename ListA::Type<Indexes>> {}, TypeWrapper<typename ListB::Type<Indexes>> {}), ...);
+    (forward<F>(f)(TypeWrapper<typename ListA::template Type<Indexes>> {}, TypeWrapper<typename ListB::template Type<Indexes>> {}), ...);
 }
 
 template<typename ListA, typename ListB, typename F>

